mne/decoding/time_gen.py

```diff
@@ -320,14 +379,18 @@ def score(self, epochs=None, y=None):
                 raise RuntimeError('Please predict() epochs first or pass '
                                    'epochs to score()')
 
-        # clean gat.score() attributes
-        for att in ['scores_', 'scorer_', 'y_true_']:
-            if hasattr(self, att):
-                delattr(self, att)
-
         # Check scorer
-        # XXX Need API to identify proper scorer from the clf
-        self.scorer_ = accuracy_score if self.scorer is None else self.scorer
+        if self.scorer is not None:
+            self.scorer_ = self.scorer
+        else:
+            if is_classifier(self.clf) and self.predict_method == 'predict':
+                self.scorer_ = accuracy_score
+            elif is_regressor(self.clf) and self.predict_method == 'predict':
+                self.scorer_ = mean_squared_error
```

Comment:
> (On the other hand, isn't every object we can use here either a regressor or a classifier?)

I'm actually not sure about this. I guess you can have cluster methods that aren't either of them. Sklearn have both `is_classifier` and `is_regressor` so let's follow their API.

> Also, I don't understand what is the purpose of the variable self.scorer_. Couldn't we use self.scorer in the repr and do away with the superfluous variable?

This is sklearn API. `scorer` without a `_` it is the param, `scorer_` is the initialized object. Typically `scorer` could be an `int`, not `scorer_`.

> So, I guess we just throw an error first if predict_method is not predict?

I don't think I follow?